pinot-query-runtime/src/test/resources/queries/Comparisons.json

```diff
@@ -0,0 +1,481 @@
+    ]
+  },
+  "between_floats": {
+    "psql": "9.2",
+    "comment": "floats silently fail (don't return any rows)",
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "val", "type": "FLOAT"},
+          {"name": "small", "type": "FLOAT"},
+          {"name": "big", "type": "FLOAT"}
+        ],
+        "inputs": [
+          [1.5, 1.1, 2.1], [1.1, 1.1, 1.1], [-1.1, -3.1, -2.1], [0.0, -1.0, 1.0], [0.0, 2147483647.0, -2147483648.0]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT val BETWEEN small AND big FROM {tbl}" },
+      { "sql": "SELECT val BETWEEN big AND small FROM {tbl}" },
+      { "sql": "SELECT val NOT BETWEEN small AND big FROM {tbl}" },
+      { "sql": "SELECT val NOT BETWEEN big AND small FROM {tbl}" }
+    ]
+  },
+  "between_doubles": {
+    "comment": "doubles silently fail (don't return any rows)",
+    "psql": "9.2",
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "val", "type": "DOUBLE"},
+          {"name": "small", "type": "DOUBLE"},
+          {"name": "big", "type": "DOUBLE"}
+        ],
+        "inputs": [
+          [1.5, 1.1, 2.1], [1.1, 1.1, 1.1], [-1.1, -3.1, -2.1], [0.0, -1.0, 1.0], [0.0, 2147483647.0, -2147483648.0]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT val BETWEEN small AND big FROM {tbl}" },
+      { "sql": "SELECT val BETWEEN big AND small FROM {tbl}" },
+      { "sql": "SELECT val NOT BETWEEN small AND big FROM {tbl}" },
+      { "sql": "SELECT val NOT BETWEEN big AND small FROM {tbl}" }
+    ]
+  },
+  "between_numerics": {
+    "comment": "big decimal silently fails (doesn't throw exception, but returns no rows)",
+    "psql": "9.2",
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "val", "type": "BIG_DECIMAL"},
+          {"name": "small", "type": "BIG_DECIMAL"},
+          {"name": "big", "type": "BIG_DECIMAL"}
+        ],
+        "inputs": [
+          ["1.5", "1.1", "2.1"], ["1.1", "1.1", "1.1"], ["-1.1", "-3.1", "-2.1"],
+          ["0.0", "-1.0", "1.0"], ["0.0", "2147483647.0", "-2147483648.0"]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT val BETWEEN small AND big FROM {tbl}" },
+      { "sql": "SELECT val BETWEEN big AND small FROM {tbl}" },
+      { "sql": "SELECT val NOT BETWEEN small AND big FROM {tbl}" },
+      { "sql": "SELECT val NOT BETWEEN big AND small FROM {tbl}" }
+    ]
+  },
+  "between_different_types": {
+    "psql": "9.2",
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "a", "type": "INT"},
+          {"name": "b", "type": "FLOAT"},
+          {"name": "c", "type": "DOUBLE"}
+        ],
+        "inputs": [
+          [1, 2.0, 3.0], [2, 3.0, 1.0], [3, 1.0, 2]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT a BETWEEN b AND c FROM {tbl}" },
+      { "sql": "SELECT b BETWEEN c AND a FROM {tbl}" },
+      { "sql": "SELECT c BETWEEN a AND b FROM {tbl}" }
+    ]
+  }
+}
```

Comment:
Can we add boolean comparison? Can we also test them in where clause?

Reply:
👍 added the boolean =/<> in the other PR, adding failure tests here and where clauses as well